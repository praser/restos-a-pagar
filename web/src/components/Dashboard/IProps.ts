interface IProps {
  anoOrcamentario: number;
  dataBloqueio: string;
}

export default IProps;
