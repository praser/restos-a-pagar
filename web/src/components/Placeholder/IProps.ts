interface IProps {
  src: string;
}

export default IProps;
