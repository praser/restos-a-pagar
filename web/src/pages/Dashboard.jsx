import React from 'react';

import { Container } from '~/components/Layout/styles';

import Navigation from '~/components/Navigation';

const Dashboard = () => {
  return (
    <Container>
      <Navigation />
    </Container>
  );
};

export default Dashboard;
