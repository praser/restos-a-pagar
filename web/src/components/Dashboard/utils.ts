import { isBefore } from 'date-fns';

import { parseISO } from 'utils/dates';
import { joinPath, locksPath, possibleLocksPath } from 'utils/paths';

import IProps from './IProps';

const getLink = (props: IProps): string | null => {
  const { anoOrcamentario, dataBloqueio } = props;
  const today = new Date();

  if (isBefore(today, parseISO(dataBloqueio) || 0))
    return joinPath(possibleLocksPath, [anoOrcamentario]);

  return joinPath(locksPath, [anoOrcamentario]);
};

export { getLink };
