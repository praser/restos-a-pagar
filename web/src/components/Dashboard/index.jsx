import { getYear, isBefore } from 'date-fns';
import { parseISO } from 'date-fns/esm';
import { first } from 'lodash';
import React, { useEffect, useState } from 'react';
import { Redirect } from 'react-router-dom';
import Layout from '~/components/Layout/Internal';
import { locksPath, joinPath, possibleLocksPath } from '~/utils/paths';
import { useApiRap } from '../../hooks';

const currYearparam = params => {
  const currYear = getYear(new Date());
  return first(params.filter(p => p.anoExecucao <= currYear));
};

const parseParam = param => {
  const {
    anoExecucao,
    anoOrcamentario,
    dataBloqueio,
    dataCancelamento,
  } = param;

  return {
    anoExecucao,
    anoOrcamentario,
    dataBloqueio: parseISO(dataBloqueio),
    dataCancelamento: parseISO(dataCancelamento),
  };
};

const getLink = params => {
  if (!params.length) return null;

  const { anoOrcamentario, dataBloqueio } = parseParam(currYearparam(params));

  const today = new Date();

  if (isBefore(today, dataBloqueio))
    return joinPath(possibleLocksPath, [anoOrcamentario]);

  return joinPath(locksPath, [anoOrcamentario]);
};

const Dashboard = () => {
  const [destiny, setDestiny] = useState();
  const apiRap = useApiRap();

  useEffect(() => {
    (async () => {
      const api = await apiRap;
      const res = await api.requests.getParams();
      setDestiny(getLink(res.data));
    })();
  }, []);

  return destiny ? <Redirect to={destiny} /> : <Layout />;
};

export default Dashboard;
