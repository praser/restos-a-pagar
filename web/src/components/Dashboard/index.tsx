import React, { useEffect, useState } from 'react';
import { Redirect } from 'react-router-dom';

import Layout from '~/components/Layout/Internal';

import IProps from './IProps';
import { getLink } from './utils';

const Dashboard = (props: IProps) => {
  const [destiny, setDestiny] = useState<string | null>();

  useEffect(() => {
    const link = getLink(props);
    setDestiny(link);
  }, [setDestiny, getLink, props]);

  return destiny ? (
    <Redirect to={destiny} />
  ) : (
    <Layout>
      <></>
    </Layout>
  );
};

export default Dashboard;
