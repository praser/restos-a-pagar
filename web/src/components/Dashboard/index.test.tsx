import React from 'react';

import renderer from 'react-test-renderer';

import IProps from './IProps';
import Dashboard from './index';

describe('Dashboard Component', () => {
  let props: IProps;
  beforeAll(() => {
    props = {
      anoOrcamentario: 2020,
      dataBloqueio: ' 2020-11-14 19:48:43.000',
    };
  });
  it('is expected to render correct component', () => {
    const dashboardComponent = renderer
      .create(<Dashboard {...props} />)
      .toJSON();
    expect(dashboardComponent).toMatchSnapshot();
  });
});
