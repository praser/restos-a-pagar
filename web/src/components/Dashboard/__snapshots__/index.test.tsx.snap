// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component is expected to render correct component 1`] = `
<div
  className="sc-kEjbxe dhrXyX"
>
  <nav
    className="sc-bdfBwQ fIcMTq"
  >
    <h1
      className="sc-gsTCUz kfVslw"
    >
      Painel de gest√£o do RAP
    </h1>
    <ul
      className="sc-dlfnbm fhAIwl"
    >
      <div
        className="sc-hKgILt hSuYkk"
      />
      <li
        className="sc-gKsewC gMZBwq"
      >
        <button
          className="sc-iBPRYJ fCMuWI"
          onClick={[Function]}
        >
          <span
            className="sc-fubCfw fDxNAs"
          />
          <img
            className="sc-pFZIQ kCLbWH"
            src="undraw_male_avatar_323b.svg"
          />
        </button>
        <div
          className="sc-eCssSg jacAYx"
        >
          <a
            className="sc-jSgupP iHQRUU"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-sign-in-alt fa-w-16 sc-jrAGrp fzTpjZ"
              data-icon="sign-in-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Sair
          </a>
        </div>
      </li>
    </ul>
  </nav>
  <div
    className="sc-iqHYGH jKTyr"
  />
</div>
`;
